codex/indexingstrategy.nim

```diff
@@ -1,12 +1,23 @@
-import std/sequtils
-
 import ./errors
 import ./utils
 import ./utils/asynciter
 
 {.push raises: [].}
 
 type
+  StrategyType* = enum
+    # Simplest approach:
+    # 0 => 0, 1, 2
+    # 1 => 3, 4, 5
+    # 2 => 6, 7, 8
+    LinearStrategy,
```

Comment:
Maybe it would be better to name it `ConsequtiveStrategy` instead of `LinearStrategy`? Examples in `SteppedStrategy` are also linear.

Reply:
Not against it, but honestly I think both names work, it's just a preference thing ;)